gap/base/methsel.gi

```diff
@@ -81,33 +88,33 @@ InstallGlobalFunction( "CallMethods", function(db, tolerancelimit, methargs...)
         i := 1;
         while i <= Length(db) do
             # skip methods which are known to be inapplicableMethods
-            if IsBound(ms.inapplicableMethods.(db[i].stamp)) then
+            if IsBound(ms.inapplicableMethods.(db[i].method.stamp)) then
                 Info(InfoMethSel, 4, "Skipping inapplicableMethods rank ", db[i].rank,
-                     " method \"", db[i].stamp, "\".");
+                     " method \"", db[i].method.stamp, "\".");
                 i := i + 1;
                 continue;
             fi;
 
             # skip methods which signalled a temporary failure at least
             # (tolerance + 1) times.
-            if IsBound(ms.failedMethods.(db[i].stamp)) and
-                ms.failedMethods.(db[i].stamp) > tolerance then
+            if IsBound(ms.failedMethods.(db[i].method.stamp)) and
+                ms.failedMethods.(db[i].method.stamp) > tolerance then
                 Info(InfoMethSel, 4, "Skipping rank ", db[i].rank,
-                     " method \"", db[i].stamp, "\".");
+                     " method \"", db[i].method.stamp, "\".");
```

Comment:
If you add methods for `Stamp` and `Comment` using `InstallOtherMethod` which take an `IsRecord`, then perhaps you could immediately change this to
```suggestion
                     " method \"", Stamp(db[i].method), "\".");
```